packages/models-tree/src/models-tree/ModelsTreeDefinition.ts

```diff
@@ -702,10 +677,13 @@ function createGeometricElementInstanceKeyPaths(
     releaseMainThreadOnItemsCount(300),
     map((row) => parseQueryRow(row, groupInfos, separator, hierarchyConfig.elementClassSpecification)),
     mergeMap(({ modelId, elementHierarchyPath, groupingNode }) =>
-      createModelInstanceKeyPaths(modelId, idsCache).pipe(
+      from(idsCache.createModelInstanceKeyPaths(modelId)).pipe(
+        mergeAll(),
         map((modelPath) => {
-          modelPath.pop(); // model is already included in the element hierarchy path
-          const path = [...modelPath, ...elementHierarchyPath];
+          // We dont want to modify the original path, we create a copy that we can modify
+          const newModelPath = [...modelPath];
+          newModelPath.pop(); // model is already included in the element hierarchy path
```

Comment:
fancier way to achieve this would be:
```ts
const [_, ...newModelPath] = modelPath;
```

feel free to resolve this comment yourself if you don't like it

Reply:
Didn't know that we can remove first element like that. But for this case it wouldn't work since `pop` removes the last element not the first